realm/realm-library/src/androidTestObjectServer/java/io/realm/SessionTests.java

```diff
@@ -188,6 +191,58 @@ public void onError(SyncSession session, ObjectServerError error) {
         SyncManager.simulateClientReset(SyncManager.getSession(config));
     }
 
+    // Check that if we manually trigger a Client Reset, then it should be possible to start
+    // downloading the Realm immediately after.
+    @Test
+    @RunTestInLooperThread
+    @Ignore("https://github.com/realm/realm-java/issues/5143")
+    public void clientReset_manualTriggerAllowSessionToRestart() {
+        SyncUser user = createTestUser();
+        String url = "realm://objectserver.realm.io/~/myrealm";
+        final AtomicReference<SyncConfiguration> configRef = new AtomicReference<>(null);
```

Comment:
why an `AtomicReference` `configRef` being final can be accessed from the runnable

Reply:
Because the `config` is being used inside the builder. You cannot reference a final field while it is being created.